test/scripts/update-kindnode-version.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 Authors of spidernet-io
+# SPDX-License-Identifier: Apache-2.0
+
+echo "Updating k8s matrix run file..."
+KIND_NODE_VERSION=""
+function update_kindnode_version() {
+    page=1
+    per_page=100
+    repo="kindest/node"
+    while true; do
+      response=$(curl -s --retry 10 "https://hub.docker.com/v2/repositories/$repo/tags/?page=$page&per_page=$per_page")
+      tags=$(echo "$response" | jq -r '.results[].name')
+      for tag in $tags; do
+        if [[ $tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          # Compatibility testing 1.22 to latest version
+          if [[ ! $KIND_NODE_VERSION =~ ${tag%.*} && ${tag%.*} > "v1.21" ]]; then
+            KIND_NODE_VERSION+="$tag, "
+          fi
+        fi
+      done
+      next=$(echo "$response" | jq -r '.next')
+      if [[ "$next" == "null" ]]; then
+          break
+      fi
+      page=$((page + 1))
+    done
+    KIND_NODE_VERSION=${KIND_NODE_VERSION}
+}
+
+update_kindnode_version
+
+KIND_NODE_VERSION="${KIND_NODE_VERSION%, }"
+KIND_NODE_VERSION="[$KIND_NODE_VERSION]"
+origin_version=$(grep -oP '(?<=version: )\[[^\]]+\]' .github/workflows/auto-diff-k8s-ci.yaml)
```

Comment:
this could be a parameter of the script , but not hard code the path

Reply:
done